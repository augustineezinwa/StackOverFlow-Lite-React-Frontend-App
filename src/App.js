import React, { Component, Fragment } from 'react';
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import NotFound from './components/notFound/NotFound';
import Footer from './components/footer/Footer';
import NavBar from './components/navbar/NavBar';
import routeTable from './routeTable';
import '../public/styles/nav.scss';
import '../public/styles/index.scss';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <Router>
        <Fragment>
          <NavBar />
          <div id="pageDisplay">
            <Switch>
              {
              routeTable.map(targetRoute => (
                <Route
                 exact={targetRoute.exact}
                 path={targetRoute.path}
                 key={targetRoute.path}
                 component={targetRoute.component}
                />
              ))
            }
              <Route component={NotFound} />
            </Switch>
          </div>
          <Footer />
        </Fragment>
      </Router>
    );
  }
}

export default App;
