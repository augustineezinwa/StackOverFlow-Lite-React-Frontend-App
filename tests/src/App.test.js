import React from 'react';
import { shallow } from 'enzyme';
import App from '../../src/App';

test('App tests', () => {
  const appComponent = shallow(<App />);
  expect(appComponent).toMatchSnapshot();
});
