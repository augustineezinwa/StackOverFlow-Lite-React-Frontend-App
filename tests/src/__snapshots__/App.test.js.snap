// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests 1`] = `
<BrowserRouter>
  <NavBar />
  <div
    id="pageDisplay"
  >
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        key="/"
        path="/"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
  <Footer />
</BrowserRouter>
`;
